
# Practicing with Genomic Data

We will use the data from this portal: https://portal.gdc.cancer.gov/

## Installing the R package

You can find use the R package here to get these data: https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html

## Getting the data

Walk through the steps of how to obtain this data


## Basic analyses

Walk through some basic analyses, referencing techniques from
https://hutchdatascience.org/Choosing_Genomics_Tools/


## Basic plots


```{r}
# Installing Your Packages
if (!require("BiocManager", quietly = TRUE))
  {install.packages("BiocManager")}


if ('TCGAbiolinks' %in% installed.packages()) {
    BiocManager::install("TCGAbiolinks")
}
# Loading packages for R to recognize
library(BiocManager)
library(TCGAbiolinks)
library(dplyr)
library(DT)
#browseVignettes("TCGAbiolinks")
# Making dataframe to store your data 
query <- GDCquery(
    project = "TCGA-BRCA",
    data.category = "Transcriptome Profiling",
    data.type = "Gene Expression Quantification",
    workflow.type = "HTSeq - Counts")
    
GDCdownload(query)

data <- GDCprepare(query)
# Exploring data 
head(data)
str(data)
summary(data)
dim(data)
colnames(data)
```

```{r}

```

```{r}
library(ggplot2)
ggplot(data, aes(x=`HTSeq - Counts`)) +
    geom_histogram(binwidth = 1) +
    ggtitle("Distribution of Gene Expression Counts") +
    xlab("Gene Expression Counts") +
    ylab("Frequency")
```

```{r}
sessionInfo()

```
